pkg/virtctl/create/instancetype/instancetype.go

```diff
@@ -0,0 +1,263 @@
+	switch c.ioThreadsPolicy {
+	case string(v1.IOThreadsPolicyAuto):
+		instancetypePolicy = v1.IOThreadsPolicyAuto
+	case string(v1.IOThreadsPolicyShared):
+		instancetypePolicy = v1.IOThreadsPolicyShared
+	default:
+		return params.FlagErr(IOThreadsPolicyFlag, "IOThread must be of value auto or shared")
+	}
+
+	instancetypeSpec.IOThreadsPolicy = &instancetypePolicy
+	return nil
+}
+
+func defaultCreateInstancetype() instancetype {
+	return instancetype{
+		name:   "instancetype-" + rand.String(5),
+		cpu:    1,
+		memory: "128Mi",
+	}
+}
+
+func (c *instancetype) usage() string {
+	return `  # Create a manifest for a Instancetype with a random name and default values:
+  {{ProgramName}} create instancetype
+  
+  # Create a manifest for a Instancetype with a specified name and cpu:
+  {{ProgramName}} create instancetype --name my-instancetype --cpu 2
+
+  # Create a manifest for a Instancetype with a specified memory and gpu:
+  {{ProgramName}} create instancetype --memory 258Mi --gpu name:gpu1,devicename:nvidia
+  
+  # Create a manifest for a ClusterInstancetype with a specified name and cpu:
+  {{ProgramName}} create instancetype --cluster --name my-instancetype --cpu 2`
+}
+
+func newInstancetype(c *instancetype) *instancetypev1alpha2.VirtualMachineInstancetype {
+	return &instancetypev1alpha2.VirtualMachineInstancetype{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "VirtualMachineInstancetype",
+			APIVersion: instancetypev1alpha2.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: c.name,
+		},
+		Spec: instancetypev1alpha2.VirtualMachineInstancetypeSpec{
+			CPU: instancetypev1alpha2.CPUInstancetype{
+				Guest: uint32(c.cpu),
+			},
+			Memory: instancetypev1alpha2.MemoryInstancetype{
+				Guest: resource.MustParse(c.memory),
+			},
+		},
+	}
+}
+
+func newClusterInstancetype(c *instancetype) *instancetypev1alpha2.VirtualMachineClusterInstancetype {
+	return &instancetypev1alpha2.VirtualMachineClusterInstancetype{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "VirtualMachineClusterInstancetype",
+			APIVersion: instancetypev1alpha2.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: c.name,
+		},
+		Spec: instancetypev1alpha2.VirtualMachineInstancetypeSpec{
+			CPU: instancetypev1alpha2.CPUInstancetype{
+				Guest: uint32(c.cpu),
+			},
+			Memory: instancetypev1alpha2.MemoryInstancetype{
+				Guest: resource.MustParse(c.memory),
+			},
+		},
+	}
+}
+
+func applyFlags(c *instancetype, cmd *cobra.Command, instancetypeSpec *instancetypev1alpha2.VirtualMachineInstancetypeSpec) error {
+	for _, flag := range flags {
+		if cmd.Flags().Changed(flag) {
+			if err := optFns[flag](c, instancetypeSpec); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func (c *instancetype) run(cmd *cobra.Command) error {
+	var out []byte
+	var err error
+
+	if c.cluster {
+		clusterInstancetype := newClusterInstancetype(c)
+		if err = applyFlags(c, cmd, &clusterInstancetype.Spec); err != nil {
```

Comment:
Why are you passing `c` here instead of having `applyFlags` as a method?

I'd also work out the overall kind of resource separately to generating the spec.